root/admin/attributes/Form.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import React from 'react';
+
+import FormRowText from '../../components/FormRowText';
+import FormSubmit from '../../components/FormSubmit';
+import FormRowSelect from '../../components/FormRowSelect';
+import FormRow from '../../components/FormRow';
+import FormRowCheckbox from '../../components/FormRowCheckbox';
+import FieldErrors from '../../components/FieldErrors';
+import FormRowTextArea from '../../components/FormRowTextArea';
+
+type LanguageFieldT = {
+  entity_type: FieldT<string | null>,
+  frequency: FieldT<number>,
+  iso_code_1: FieldT<string | null>,
+  iso_code_2b: FieldT<string | null>,
+  iso_code_2t: FieldT<string | null>,
+  iso_code_3: FieldT<string | null>,
+  name: FieldT<string>,
+};
+
+type ScriptFieldT = {
+  entity_type: FieldT<string | null>,
+  frequency: FieldT<number>,
+  iso_code: FieldT<string | null>,
+  iso_number: FieldT<string | null>,
+  name: FieldT<string>,
+};
+
+type Props = {
+  form: FormT<LanguageFieldT> | FormT<ScriptFieldT>,
+  id: number,
+  model: string,
```

Comment:
About the Flow issues you mentioned on IRC:

You need a way to have Flow associate the model here with the type of form. The primary way to do that is by using a union type for the props:

```JS
type BaseProps = {id: number, parentOptions: ...};

type Props =
  | {form: FormT<LanguageFieldT>, model: 'Language', ...BaseProps}
  | {form: FormT<ScriptFieldT>, model: 'Script', ...BaseProps}
  ;
```

You can then use a [disjoin union refinement](https://flow.org/en/docs/types/unions/#toc-disjoint-unions) below. A limitation of Flow is that this only works if you reference `props.model` in the switch and `props.form` for the form (don't destructure them in the parameter list), though you can save the latter to a variable after the refinement is made.

Reply:
@mwiencek I was trying something like this 

> `type Props =
>   | {
>     form: FormT<LanguageFieldT>,
>     id: number,
>     model: 'Language',
>     parentOptions: $ReadOnlyArray<{label: string, value: number}>}
>   | {
>     form: FormT<ScriptFieldT>,
>     id: number,
>     model: 'Script',
>     parentOptions: $ReadOnlyArray<{label: string, value: number}>};`

But still getting errors and also getting error for model now. 
<img width="664" alt="Screenshot 2019-05-11 at 4 13 06 PM" src="https://user-images.githubusercontent.com/14249589/57568625-bd3d1200-7407-11e9-891f-4e238c3f6e64.png">
And below is the previous type issue I had 
<img width="662" alt="Screenshot 2019-05-11 at 4 14 18 PM" src="https://user-images.githubusercontent.com/14249589/57568635-e5c50c00-7407-11e9-81ca-7a8de4d33880.png">